src/appleseed.studio/mainwindow/renderingsettingswindow.cpp

```diff
@@ -707,6 +747,35 @@ namespace
 
             set_config(config, param_path, max_path_length);
         }
+
+        void load_separate_bounce_settings(
+            const Configuration&    config,
+            const string&           widget_key_prefix,
+            const string&           param_path,
+            const string&           type)
+        {
+            const size_t DefaultMaxBounces = 8;
+
+            const size_t max_path_length =
+                get_config<size_t>(config, param_path, 0);
+
+            set_widget(widget_key_prefix + ".bounces.unlimited_"+type+"_bounces", max_path_length == 0);
+            set_widget(widget_key_prefix + ".bounces.max_"+type+"_bounces", max_path_length == 0 ? DefaultMaxBounces : max_path_length - 1);
```

Comment:
The `- 1` here is probably incorrect. It used to exist to convert path length to bounces. But your case is simpler as there should not be any conversion, you're only dealing with number of bounces.